content/doc/tutorials/build-a-python-app-with-pyinstaller.adoc

```diff
@@ -409,147 +315,74 @@ pipeline {
+Although this step consists of a single command, as a general principle, it's a good idea to keep your Pipeline code (i.e. the `Jenkinsfile`) as tidy as possible and place more complex build steps (particularly for stages consisting of 2 or more steps) into separate shell script files.
+This ultimately makes maintaining your Pipeline code easier, especially if your Pipeline gains more complexity.
+<3> This link:/doc/pipeline/steps/core/#archiveArtifacts-archive-the-artifacts[`archiveArtifacts`] step (provided as part of Jenkins core) archives the standalone executable file (generated by the `pyinstaller` command above at `dist/add2vals` within the Jenkins home's workspace directory) and exposes this file through the Jenkins interface.
+The link:/doc/book/pipeline/syntax/#post[`post`] section's `success` condition that contains this `archiveArtifacts` step ensures that the step is executed _at the completion_ of the `Deliver` stage _only if_ this stage completed successfully.
 . Save your edited `Jenkinsfile` and commit it to your local
   `simple-python-pyinstaller-app` Git repository. E.g. Within the
   `simple-python-pyinstaller-app` directory, run the commands: +
-  `git stage .` +
+  `git add .` +
   then +
-  `git commit -m "Add 'Deliver' stage"`
-. Go back to Jenkins again, log in again if necessary and ensure you've accessed
-  Jenkins's Blue Ocean interface.
-. Click *Run* at the top left, then quickly click the *OPEN* link which appears
-  briefly at the lower-right to see Jenkins running your amended Pipeline
-  project. If you weren't able to click the **OPEN** link, click the _top_ row
-  on the Blue Ocean interface to access this feature. +
-  **Note:** It may take a few minutes for the `cdrx/pyinstaller-linux` Docker
-  image to download (if this hasn't already been done). +
-  If your amended Pipeline ran successfully, here's what the Blue Ocean
-  interface should look like. Notice the additional "Deliver" stage. Click on
-  the previous "Test" and "Build" stage circles to access the outputs from those
-  stages.
+  `git commit -m "Add 'Deliver' stage"` +
+and in the end +
+`git push`
+
+. In Jenkins, sign in if necessary, and go back to the *Dashboard* and then *simple-python-pyinstaller-app* or go directly to *simple-python-pyinstaller-app* depending on where you're starting from.
+. Select *Build Now* on the left. You should see after a while a new column *Deliver* appear in the *Stage View*.
 +
 [.boxshadow]
-image:tutorials/python-pyinstaller-21-deliver-stage-runs-successfully.png[alt="Deliver
-stage runs successfully",width=100%]
+image:tutorials/python/deliver-stage-added.png[alt="Deliver stage runs successfully",width=100%]
+
+. You can click on *#3* or on the number representing your last build on the left, under *Build History*. You will then see the details of the build.
```

Comment:
```suggestion
. You can click on *#3* or on the number representing your last build, in the left pane under *Build History* to see the details of the build.
```
Suggestion to adjust wording so that it is one sentence and describe the location clearly.